src/plugin/protocol-smb/src/java/org/apache/nutch/protocol/smb/Smb.java

```diff
@@ -0,0 +1,292 @@
+ */
+package org.apache.nutch.protocol.smb;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.nutch.crawl.CrawlDatum;
+import org.apache.nutch.metadata.Metadata;
+import org.apache.nutch.net.protocols.HttpDateFormat;
+import org.apache.nutch.protocol.Content;
+import org.apache.nutch.protocol.Protocol;
+import org.apache.nutch.protocol.ProtocolOutput;
+import org.apache.nutch.protocol.ProtocolStatus;
+import org.apache.nutch.protocol.RobotRulesParser;
+import com.hierynomus.msdtyp.AccessMask;
+import com.hierynomus.msfscc.FileAttributes;
+import com.hierynomus.msfscc.fileinformation.FileAllInformation;
+import com.hierynomus.msfscc.fileinformation.FileIdBothDirectoryInformation;
+import com.hierynomus.mssmb2.SMB2CreateDisposition;
+import com.hierynomus.mssmb2.SMB2CreateOptions;
+import com.hierynomus.mssmb2.SMB2ShareAccess;
+import com.hierynomus.smbj.auth.AuthenticationContext;
+import com.hierynomus.smbj.connection.Connection;
+import com.hierynomus.smbj.session.Session;
+import com.hierynomus.smbj.share.DiskShare;
+import com.hierynomus.smbj.share.File;
+import com.hierynomus.smbj.SMBClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import crawlercommons.robots.BaseRobotRules;
+
+public class Smb implements Protocol {
+  protected static final Logger LOG = LoggerFactory.getLogger(Smb.class);
+
+  private Configuration conf;
+
+  private String user;
+  private String password;
+  private String domain;
+  private int contentLimit;
+  private Set<String> ignoreFiles;
+
+  public Smb() {
+    // todo: files that should be skipped could be configurable.
+    this.ignoreFiles = new HashSet<>();
+    ignoreFiles.add(".");
+    ignoreFiles.add("..");
+    ignoreFiles.add(".svn");
+    ignoreFiles.add(".git");
+  }
+
+  @Override
+  public Configuration getConf() {
+    LOG.debug("getConf()");
+    return this.conf;
+  }
+
+  @Override
+  public void setConf(Configuration conf) {
+    this.conf = conf;
+
+    // todo: is it possible to use configuration "per server" or "per share"?
+    user = conf.getTrimmed("smb.user");
+    if (user == null || user.isEmpty()) {
+      throw new IllegalArgumentException("Config parameter 'smb.user' not set.");
+    }
+    password = conf.getTrimmed("smb.password");
+    if (password == null || password.isEmpty()) {
+      throw new IllegalArgumentException("Config parameter 'smb.password' not set.");
+    }
+    domain = conf.getTrimmed("smb.domain");
+    contentLimit = conf.getInt("smb.content.limit", Integer.MAX_VALUE);

```

Comment:
Please take a look at [http.content.limit](https://github.com/apache/nutch/blob/master/conf/nutch-default.xml#L255) and [file.content.limit](https://github.com/apache/nutch/blob/master/conf/nutch-default.xml#L511)... I would suggest we mimic those values for now.